generators/new_collateral/config/parser_test.go

```diff
@@ -34,107 +33,55 @@ var _ = Describe("Config Parser", func() {
-				Exporter: map[string]interface{}{
-					"home": "home", "name": "name", "save": true, "schema": "schema",
-				},
-			}
-			exporterMetadata, parseErr := config.ParseExporterMetaData(tomlConfig)
+			configFileName := "testConfigWithNoTransformerNames"
+			config, parseErr := configParser.ParseCurrentConfig(testConfigFilePath, configFileName)
 			Expect(parseErr).NotTo(HaveOccurred())
-			Expect(exporterMetadata).To(Equal(types.ExporterMetaData{
-				Home:             "home",
-				Name:             "name",
+			Expect(config.ExporterMetadata).To(Equal(types.ExporterMetaData{
+				Home:             "github.com/makerdao/vulcanizedb",
+				Name:             "transformerExporter",
 				Save:             true,
-				Schema:           "schema",
+				Schema:           "maker",
 				TransformerNames: nil,
 			}))
 		})
-	})
 
-	Context("ParseTransformerExporters", func() {
 		It("parses transformerExporters", func() {
-			transformerExporterMap := map[string]interface{}{
-				"path":       "path",
-				"type":       "type",
-				"repository": "repository",
-				"migrations": "migrations",
-				"contracts":  []string{"testContract"},
-				"rank":       "0",
-			}
-			tomlConfig := types.TransformersConfigForToml{
-				Exporter: map[string]interface{}{
-					"test-exporter": transformerExporterMap,
-				},
-			}
-			transformerExporters, parseErr := config.ParseTransformerExporters(tomlConfig)
+			config, parseErr := configParser.ParseCurrentConfig(testConfigFilePath, testConfigFileName)
 			Expect(parseErr).NotTo(HaveOccurred())
-			Expect(transformerExporters).To(Equal(types.TransformerExporters{
-				"test-exporter": {
-					Path:       "path",
-					Type:       "type",
-					Repository: "repository",
-					Migrations: "migrations",
-					Contracts:  []string{"testContract"},
-					Rank:       "0",
-				},
-			}))
+			Expect(config.TransformerExporters).To(Equal(expectedParsedConfig.TransformerExporters))
 		})
 
 		It("returns an error if it fails to decode an exporterValue", func() {
-			// this map will not properly decode into a types.TransformerExporter because the Path field doesn't match
-			notATransformerExporterMap := map[string]interface{}{
-				"Path": 1,
-			}
-			tomlConfig := types.TransformersConfigForToml{
-				Exporter: map[string]interface{}{
-					"test-exporter": notATransformerExporterMap,
-				},
-			}
-			_, parseErr := config.ParseTransformerExporters(tomlConfig)
+			// the exporter.cat_v1_1_0 will not properly decode into a types.TransformerExporter because the Path field doesn't match
+			configFile := "testConfigWithBadTransformerExporter"
+			_, parseErr := configParser.ParseCurrentConfig(testConfigFilePath, configFile)
 			Expect(parseErr).To(HaveOccurred())
-			Expect(parseErr.Error()).To(MatchRegexp("'Path' expected type 'string', got unconvertible type 'int'"))
+			Expect(parseErr).To(MatchError(
+				"error parsing transformer exporters from config file: error decoding transformerExporters:" +
```

Comment:
I was struggling with how best to test these wrapped errors - would love some help figuring out a better way to test them!

Reply:
This is maybe kind of cheating and idk if we may want to match against the nested error, but if not - maybe we could create an error to wrap the description and then match against that in the tests, e.g.

in the source
```
var ErrParsingConfig = errors.Net("error parsing transformer exporters from config file")
fmt.Errorf("%w: %s", ErrParsingConfig, transformerExportersErr.Error())
```

in the tests:
```
Expect(parseErr).To(MatchError(ErrParsingConfig))
```

After some quick reading, it doesn't seem like there's a simple way to wrap multiple errors with `fmt.Errorf` and `%w` :/

Reply:
Good idea - made the change in ecd4f99